### PR TITLE
draft version of public api doc

### PR DESCRIPTION
**Pull request by the-mx**

### CODE REVIEW

File: docs/public_api.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+Draft:
+Our subnet allows validators to run serve http (websocket) endpoint that allows external (external to subnet) clients to access the power (probably power is not the best word) of the subnet.
```
Comment (rhohndorf) [inline docs/public_api.md:2]: external to the subnet

Comment (rhohndorf) [reply docs/public_api.md:2]: instead of power maybe
* services provided by the subnet

File: docs/public_api.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+# Short version
+To serve public endpoint you need to:
```
Comment (rhohndorf) [inline docs/public_api.md:5]: to serve a public endpoint

File: docs/public_api.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+1. Add `--public_api.enabled` to `neurons/validator.config.js`.
+2. Add `--public_api.server_port 8888` (default value) to `neurons/validator.config.js` and open the port.
+3. Add client API key with `scripts/api_key_cli.sh ~/.bittensor/miners/validator/default/netuid17/validator/api_keys.db add EWJzrqUL94r5BEz0ZLPUbEwD8qdQX18dCUF9B7C2e0 client-name 100 60`
+4. Wait for 30 minutes.
+
+# Configuring the validator 
+To start serving public endpoint, you need to turn it on. You need to run validator with `--public_api.enabled` flag.
```
Comment (rhohndorf) [inline docs/public_api.md:12]: To start serving a public endpoint

Comment (rhohndorf) [reply docs/public_api.md:12]: You need to run the validator with `--public_api.enabled` flag.

File: docs/public_api.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+By default `--public_api.server_port` is set to 8888, you can change it and open corresponding port on your firewall.
```
Comment (rhohndorf) [inline docs/public_api.md:13]: open the corresponding...

File: docs/public_api.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+Other public API settings:
+`--public_api.copies` - validator will give the task to `--public_api.copies` miners and pick the best results. It's not recommended to change this settings. Default value 4 is considered a good balance between  latency and redundancy.
```
Comment (rhohndorf) [inline docs/public_api.md:16]: The validator will assign the task...

Comment (rhohndorf) [reply docs/public_api.md:16]: By default four miners are assigned to each task, which has shown to be a good balance between latency and redundancy

File: docs/public_api.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+`--public_api.wait_after_first_copy` - once the first results are received, validator will wait for `--public_api.wait_after_first_copy` seconds to let other miners to finish with their task. It's not recommended to change this value. More data will be collected and this parameter will be adjusted in a future releases.
```
Comment (rhohndorf) [inline docs/public_api.md:17]: Once..., the validator will... to let the remaining miners finish...

File: docs/public_api.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+./api_key_cli.sh ~/.bittensor/miners/validator/default/netuid17/validator/api_keys.db list
+```
+
+## Add new API key
+To add new client API key you need to provide some client name or alias, and specify rate limit parameters (number of requests per defined period).
+The first parameter defines the number of requests, the second interval in seconds.
+```commandline
+./api_key_cli.sh ~/.bittensor/miners/validator/default/netuid17/validator/api_keys.db add EWJzrqUL94r5BEz0ZLPUbEwD8qdQX18dCUF9B7C2e0 client-name 100 60
+```
+In this example, client would be allowed to do 100 requests per 60 seconds (1 minute).
```
Comment (rhohndorf) [inline docs/public_api.md:45]: the client is allowed to generate ...

File: docs/public_api.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+2. **Miner 1 Interactions**:
+    - `Miner 1` pulls a task from `Validator`.
+    - The `Validator` sends the task details to `Miner 1`.
+    - The `Validator` sends a task update (`STARTED`) to the `Client`.
+
+3. **Miner 2 Interactions**:
+    - `Miner 2` pulls a task from `Validator`.
+    - The `Validator` sends the task details to `Miner 2`.
+
```
Comment (teshaTe) [inline docs/public_api.md:95]: missing:   - The `Validator` sends a task update (`STARTED`) to the `Client`
yo make both miner1 and miner2 conistent

**Review by rhohndorf — APPROVED** [review]

**Pull request merged**